google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriter.java

```diff
@@ -0,0 +1,128 @@
+
+/** A helper class that adds dependencies of Cloud libraries to a given module. */
+final class CloudLibraryDependencyWriter {
+
+  /** Prevents instantiation. */
+  private CloudLibraryDependencyWriter() {}
+
+  /**
+   * Adds the given set of {@link CloudLibrary CloudLibraries} to the given {@link Module}.
+   *
+   * <p>For a module whose project manages its dependencies through Maven, the list of libraries
+   * will be added as dependencies to the {@code pom.xml}. For all other dependency management
+   * systems, the libraries will be downloaded and added directly to the module's classpath.
+   *
+   * @param libraries the set of {@link CloudLibrary CloudLibraries} to add
+   * @param module the {@link Module} to add the libraries to
+   */
+  static void addLibraries(@NotNull Set<CloudLibrary> libraries, @NotNull Module module) {
+    if (libraries.isEmpty()) {
+      return;
+    }
+
+    Project project = module.getProject();
+    MavenProjectsManager projectsManager = MavenProjectsManager.getInstance(module.getProject());
+    MavenProject mavenProject = projectsManager.findProject(module);
+    if (mavenProject == null) {
+      // TODO(nkibler): Handle non-Maven projects.
+      return;
+    }
+
+    MavenDomProjectModel model =
+        MavenDomUtil.getMavenDomProjectModel(project, mavenProject.getFile());
+    if (model == null) {
+      // TODO(nkibler): Handle this error-state, maybe a warning?
+      return;
+    }
+
+    new WriteCommandAction(project, DomUtil.getFile(model)) {
+      @Override
+      protected void run(@NotNull Result result) throws Throwable {
+        List<MavenDomDependency> dependencies = model.getDependencies().getDependencies();
+        // TODO(nkibler): Inform the user when libraries were ignored because of existing
+        // dependencies, maybe a warning?
+        libraries
+            .stream()
+            .map(CloudLibraryUtils::getFirstJavaClientMavenCoordinates)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(CloudLibraryDependencyWriter::toMavenId)
+            .filter(mavenId -> isMavenIdNotInDependencyList(mavenId, dependencies))
+            .forEach(
+                mavenId -> MavenDomUtil.createDomDependency(model, /* editor= */ null, mavenId));
+      }
+    }.execute();
+  }
+
+  /**
+   * Returns {@code true} if the given {@link MavenId} is not in the given list of {@link
+   * MavenDomDependency dependencies}.
+   *
+   * <p>Note that equality is tested via matching group IDs and artifact IDs. Equality of versions
+   * is not required. This prevents adding duplicate dependencies for the same artifact, but with a
+   * different version.
+   *
+   * @param mavenId the {@link MavenId} to check for existence in the given list of dependencies
+   * @param dependencies the list of {@link MavenDomDependency} objects that currently exist in the
+   *     DOM model
+   */
+  private static boolean isMavenIdNotInDependencyList(
```

Comment:
ditto with informing the user if this is the case so the understand what the dependency wasn't added

Reply:
Done.